### DISCUSSION TITLE
小引擎的线程模型是怎样的？

### DISCUSSION THREAD

**Discussion by Chinouo** [BoomingTech/Piccolo#330] [Q&A] [answered]
最近对多线程感兴趣，但是不知道现代引擎是如何设计线程和分配任务的，想了解一下。

**Comment by Ol6rin** [comment, marked as answer]
Piccolo目前是单线程的，tick Level时逐个tick GObject的各个Component。
你的问题我们最新的一节课程刚好能够回答。欢迎到B站收看回放视频。

20.现代游戏引擎架构：面向数据编程与任务系统（上）
https://www.bilibili.com/video/BV1EP411V7jx?spm_id_from=333.337.search-card.all.click&vd_source=f12a5db552661d28e8507875c37983cd

20.现代游戏引擎架构：面向数据编程与任务系统（下）
https://www.bilibili.com/video/BV1Md4y1G7zp/?spm_id_from=333.788&vd_source=f12a5db552661d28e8507875c37983cd
